content/powershell/concepts/objects/objects.md

```diff
@@ -1,13 +1,91 @@
 ---
 Title: 'Objects'
-Description: ''
+Description: 'Objects are made up of properties and methods. PowerShell allows us to create custom objects and add our own variables and functions to them.'
 Subjects:
   - 'Computer Science'
+  - 'PowerShell'
 Tags:
   - 'Objects'
+  - 'OOP'
+  - 'Classes'
   - 'Data Types'
   - 'Data Structures'
 CatalogContent:
   - 'learn-powershell'
   - 'paths/computer-science'
 ---
+
+Everything in PowerShell is an **object**. An object is a combination of variables and functions.  Each object has:
+- **Properties**: variables that describe the object (characteristics)
+- **Methods**: functions that describe how to use the object (actions)
+
+Each object is an instance of a blueprint called a *type*, or *[class](https://en.wikipedia.org/wiki/Class_(computer_programming))*. 
+
+## Getting Properties and Methods of an Object
+All information associated with an object is called a *member*. The `Get-Member` cmdlet reveals the member of an object – properties, methods, and so on. It also shows the type of an object.
+```PowerShell
+$dog | Get-Member
+```
+
+## Object Properties
+Properties, or attributes, are characteristics of an object. String objects, for example, have a property called `Length` which holds the number of characters in a string.
+```PowerShell
+PS > "hello".Length # "hello" has 5 characters
+5
+```
+
+The `Get-Member` cmdlet can be utilized with the `MemberType` parameter to only show the properties of an object. Consider a script file:
+```PowerShell
+Get-ChildItem .\script.ps1 | Get-Member -MemberType Property # Shows all properties of the script
+(Get-ChildItem .\script.ps1).CreationTime # Prints the date when the script was created
+```
+
+## Object Methods
+Methods are the actions an object can take. One of the methods of a String object, for example, is `Contains()`. This method returns `True` if a string contains the specified substring, otherwise `False`.
```

Comment:
```suggestion

Methods are the actions an object can take. One of the methods of a String object, for example, is `Contains()`. This method returns `True` if a string contains the specified substring, otherwise `False`.

```